apps/src/templates/manageStudents/manageStudentsRedux.js

```diff
@@ -8,19 +8,39 @@ const CANCEL_EDITING_STUDENT = 'manageStudents/CANCEL_EDITING_STUDENT';
 const REMOVE_STUDENT = 'manageStudents/REMOVE_STUDENT';
 const SET_SECRET_IMAGE = 'manageStudents/SET_SECRET_IMAGE';
 const SET_SECRET_WORDS = 'manageStudents/SET_SECRET_WORDS';
+const EDIT_NAME = 'manageStudents/EDIT_NAME';
+const EDIT_AGE = 'manageStudents/EDIT_AGE';
+const EDIT_GENDER = 'manageStudents/EDIT_GENDER';
+const START_SAVING_STUDENT = 'manageStudents/START_SAVING_STUDENT';
+const SAVE_STUDENT_SUCCESS = 'manageStudents/SAVE_STUDENT_SUCCESS';
 
 export const setLoginType = loginType => ({ type: SET_LOGIN_TYPE, loginType });
 export const setSectionId = sectionId => ({ type: SET_SECTION_ID, sectionId});
 export const setStudents = studentData => ({ type: SET_STUDENTS, studentData });
 export const startEditingStudent = (studentId) => ({ type: START_EDITING_STUDENT, studentId });
 export const cancelEditingStudent = (studentId) => ({ type: CANCEL_EDITING_STUDENT, studentId });
+export const saveStudentSuccess = (studentId) => ({ type: SAVE_STUDENT_SUCCESS, studentId });
 export const removeStudent = (studentId) => ({ type: REMOVE_STUDENT, studentId });
-export const setSecretImage = (studentId, image) => ({ type: SET_SECRET_IMAGE, studentId, image});
-export const setSecretWords = (studentId, words) => ({ type: SET_SECRET_WORDS, studentId, words});
+export const setSecretImage = (studentId, image) => ({ type: SET_SECRET_IMAGE, studentId, image });
+export const setSecretWords = (studentId, words) => ({ type: SET_SECRET_WORDS, studentId, words });
+export const editName = (studentId, name) => ({ type: EDIT_NAME, studentId, name });
+export const editAge = (studentId, age) => ({ type: EDIT_AGE, studentId, age });
+export const editGender = (studentId, gender) => ({ type: EDIT_GENDER, studentId, gender });
+
+export const startSavingStudent = (studentId) => {
```

Comment:
As an exported action creator, I would just name this `saveStudent` - from the outside world's point of view, this is a single asynchronous operation.